src/gui/creds/flow2auth.cpp

```diff
@@ -0,0 +1,164 @@
+	//       message by the server (security, intended design).
+    openBrowser();
+}
+
+QUrl Flow2Auth::authorisationLink() const
+{
+    return _loginUrl;
+}
+
+void Flow2Auth::openBrowser()
+{
+    _pollTimer.stop();
+
+    // Step 1: Initiate a login, do an anonymous POST request
+    QUrl url = Utility::concatUrlPath(_account->url().toString(), QLatin1String("/index.php/login/v2"));
+
+    auto job = _account->sendRequest("POST", url);
+    job->setTimeout(qMin(30 * 1000ll, job->timeoutMsec()));
+
+    QObject::connect(job, &SimpleNetworkJob::finishedSignal, this, [this](QNetworkReply *reply) {
+        auto jsonData = reply->readAll();
+        QJsonParseError jsonParseError;
+        QJsonObject json = QJsonDocument::fromJson(jsonData, &jsonParseError).object();
+
+        QString pollToken = json.value("poll").toObject().value("token").toString();
+        QString pollEndpoint = json.value("poll").toObject().value("endpoint").toString();
+        QUrl loginUrl = json["login"].toString();
+
+        if (reply->error() != QNetworkReply::NoError || jsonParseError.error != QJsonParseError::NoError
+            || json.isEmpty() || pollToken.isEmpty() || pollEndpoint.isEmpty() || loginUrl.isEmpty()) {
+            QString errorReason;
+            QString errorFromJson = json["error"].toString();
+            if (!errorFromJson.isEmpty()) {
+                errorReason = tr("Error returned from the server: <em>%1</em>")
+                                  .arg(errorFromJson.toHtmlEscaped());
+            } else if (reply->error() != QNetworkReply::NoError) {
+                errorReason = tr("There was an error accessing the 'token' endpoint: <br><em>%1</em>")
+                                  .arg(reply->errorString().toHtmlEscaped());
+            } else if (jsonParseError.error != QJsonParseError::NoError) {
+                errorReason = tr("Could not parse the JSON returned from the server: <br><em>%1</em>")
+                                  .arg(jsonParseError.errorString());
+            } else {
+                errorReason = tr("The reply from the server did not contain all expected fields");
+            }
+            qCWarning(lcFlow2auth) << "Error when getting the loginUrl" << json << errorReason;
+            emit result(Error);
+            return;
+        }
+
+
+        _loginUrl = loginUrl;
+        _pollToken = pollToken;
+        _pollEndpoint = pollEndpoint;
+
+
+        ConfigFile cfg;
+        std::chrono::milliseconds polltime = cfg.remotePollInterval();
+        qCInfo(lcFlow2auth) << "setting remote poll timer interval to" << polltime.count() << "msec";
+        _pollTimer.setInterval(polltime.count());
+        QObject::connect(&_pollTimer, &QTimer::timeout, this, &Flow2Auth::slotPollTimerTimeout);
+        _pollTimer.start();
+
+
+		if (!QDesktopServices::openUrl(authorisationLink())) {
+            // TODO: Ask the user to copy and open the link instead of failing here!
```

Comment:
Right good point!

Reply:
UI handles it now (see https://github.com/nextcloud/desktop/pull/1384/commits/24aeb56d0d9be6b65ceb80db9bce667f6e32fdd1) 😼